components/events/utils.ts

```diff
@@ -0,0 +1,68 @@
+import { ref } from "vue";
+
+export interface Event {
+  id: number;
+  name: string;
+  location: string;
+  summary: string;
+  description: string;
+  slides: string;
+  organizer: string;
+  startTime: string;
+  endTime: string;
+}
+
+export let all_events: Ref<Event[]> = ref([]);
+
+export function getAllEvents() {
+  fetch("/api/events/all")
+    .then(response => {
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+      return response.json();
+    })
+    .then(data => {
+      all_events.value = data;
+    })
+    .catch(error => {
+      console.error("There was a problem with the fetch operation:", error);
+    });
+}
+
+
+export function getEvent(id: string, callback: (data: Event | null, error: Error | null) => void) {
```

Comment:
why exacty are we entering callback hell here? Why dont you just return the value and every consumer of the function can just call `.then().then().....`

Reply:
you could basically just replace the whole thing with `return fetch('....').then(responsechecking)`